spec/absolute.html

```diff
@@ -2,103 +2,374 @@
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal.absolute.fromepochnanoseconds">
+      <h1>Temporal.Absolute.fromEpochNanoseconds ( _arg_ )</h1>
+      <p>
+        The `fromEpochNanoseconds` method takes one argument _arg_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _epochNanoseconds_ be ? ToBigInt(_arg_).
+        1. Return ? CreateAbsolute(_epochNanoseconds_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal.absolute.compare">
+      <h1>Temporal.Absolute.compare ( _one_, _two_ )</h1>
+      <p>
+        The `compare` method takes two arguments, _one_ and _two_.
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _one_ be ? ToAbsolute(_one_).
+        1. Let _two_ be ? ToAbsolute(_two_).
+        1. Return ! AbsoluteCompare(_one_, _two_).
+      </emu-alg>
+    </emu-clause>
   </emu-clause>
-  <emu-clause id="sec-temporal-absolute.fromEpochNanoseconds">
-    <h1>Absolute.fromEpochNanoseconds(_nanoseconds_)</h1>
-    <p>
-      This function will create an Absolute object that represents the absolute in time _nanoseconds_ since epoch such that
-      `absolute.epochNanoseconds === absolute.epochNanoseconds` and `absolute.epochMicroseconds === _microseconds_`
-      are both `true`.
-    </p>
+
+  <emu-clause id="sec-properties-of-the-temporal-absolute-prototype-object">
+    <h1>Properties of the Temporal.Absolute Prototype Object</h1>
+
+    <p>The Temporal.Absolute prototype object</p>
+    <ul>
+      <li>is the intrinsic object %Temporal.Absolute.prototype%.
+      <li>is itself an ordinary object.
+      <li>is not a Temporal.Absolute instance and does not have a [[InitializedTemporalAbsolute]] internal slot.
+      <li>has a [[Prototype]] internal slot whose value is %Object.prototype%.
+    </ul>
+
+    <emu-clause id="sec-temporal.Absolute.prototype.constructor">
+      <h1>Temporal.Absolute.prototype.constructor</h1>
+      <p>The initial value of Temporal.Absolute.prototype.constructor is %Temporal.Absolute%.</p>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal.Absolute.prototype-@@tostringtag">
+      <h1>Temporal.Absolute.prototype[ @@toStringTag ]</h1>
+      <p>
+        The initial value of the @@toStringTag property is the string value `"Temporal.Absolute"`.
+      </p>
+      <p>
+        This property has the attributes { [[Writable]]: *false*, [[Enumerable]]: *false*, [[Configurable]]: *true* }.
+      </p>
+    </emu-clause>
+
+    <emu-clause id="sec-temporal.absolute.prototype.getepochseconds">
+      <h1>Temporal.Absolute.prototype.getEpochSeconds ( )</h1>
+      <p>
+        The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _absolute_ be the *this* value.
+        1. Perform ? RequireInternalSlot(_absolute_, [[InitializedTemporalAbsolute]]).
+        1. Let _ns_ be _absolute_.[[Nanoseconds]].
+        1. Let _s_ be floor(_ns_ / 1,000,000,000).
```

Comment:
Integer maths has no `floor`. And for negative numbers `floor` would be wrong. So that step is superfluous. 

`_absolute_.[[Nanoseconds]]` is a 64bit integer.

Reply:
Math in the JS spec was "mathematical values" by default. Now it's Numbers by default, but you can opt into BigInts if you want. I think this calculation is correct, since _ns_ will be positive anyway.

Reply:
@littledan OK, in that case we need to specify integer maths.

And **NO this will not be always positive**. The ns value here would be negative for anything before the epoch. (ie: 1969-12-31T23:59:59.999999999Z has ns value of -1n)

Reply:
So, what's actually the rounding we want? Towards zero? Round to nearest (and what with .5? Bankers' rounding?) Or do we want to return fractions in some of these?

Reply:
Round toward 0

Reply:
See the JS spec's mathematical notation in https://tc39.es/ecma262/#sec-mathematical-operations . We could add more (harder) or we could use some awkward expression in words to describe exactly what we want here (easier).